### ISSUE TITLE
Rule: Spaces inside of parens

### ISSUE THREAD

**Issue by nzakas** [eslint/eslint#627]
We need a rule that enforces or disallows spaces inside of parens. It should accept one option, "always" to enforce spaces inside of parens:

``` js
foo( a, b );
```

And "never", which would enforce no spaces:

``` js
foo(a, b);
```

For both options, this is okay:

``` js
foo();
```

The rule ID should be `space-in-parens` and should be on with "never" as the default option.
